[Suggestion] Alternative for per-user NPM configuration so that `claude` does not want user to install into `/usr` as superuser.

I'm not a Node developer; [asked Claude for best way to install the package as a non-root user](https://claude.ai/share/36c35f55-28c3-4067-bc82-1d3f9fa144b5).

Error:
```text
npm error   errno: -13,
npm error   code: 'EACCES',
npm error   syscall: 'mkdir',
npm error   path: '/usr/lib/node_modules/@anthropic-ai/claude-code'
npm error }
npm error
npm error The operation was rejected by your operating system.
npm error It is likely you do not have the permissions to access this file as the current user
npm error
npm error If you believe this might be a permissions issue, please double-check the
npm error permissions of the file and its containing directories, or try running
npm error the command again as root/Administrator.
```

Was able to successfully install as a normal user following Claude's advice.

Could consider a security warning for other people who might also be unfamiliar with Node.js/NPM and care about running as a non-root user.

I'll second this. On Ubuntu 22.04 & npm v10.5.0, I get the following error because it tries to install to `/usr/lib/node_modules`.

```
$ npm install -g @anthropic-ai/claude-code
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/lib/node_modules/@anthropic-ai
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/lib/node_modules/@anthropic-ai'
...
```

I'd rather not use sudo / install this system-wide. Thanks @emcd for the solution:

```
mkdir ~/.npm-global
npm config set prefix '~/.npm-global'
export PATH=~/.npm-global/bin:$PATH         # add this to your bashrc!
npm install -g @anthropic-ai/claude-code
```

I was also not able to do this.

following the suggestion first:
```zsh
> sudo chown -R $USER:$(id -gn) /usr/local && sudo chmod -R u+w /usr/local
Password: <i enter my password>
chown: /usr/local: Operation not permitted
```

then trying the second suggestion:

```zsh
> npm -g config set prefix /Users/oliclive-griffin/.npm-global
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/local/etc
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/local/etc'
npm ERR!     at async mkdir (node:internal/fs/promises:859:10)
npm ERR!     at async Config.save (/usr/local/lib/node_modules/npm/node_modules/@npmcli/config/lib/index.js:771:5)
npm ERR!     at async Config.set (/usr/local/lib/node_modules/npm/lib/commands/config.js:176:5)
npm ERR!     at async Config.exec (/usr/local/lib/node_modules/npm/lib/commands/config.js:118:11)
npm ERR!     at async module.exports (/usr/local/lib/node_modules/npm/lib/cli-entry.js:61:5)
npm ERR!  Error: EACCES: permission denied, mkdir '/usr/local/etc'
npm ERR!     at async mkdir (node:internal/fs/promises:859:10)
npm ERR!     at async Config.save (/usr/local/lib/node_modules/npm/node_modules/@npmcli/config/lib/index.js:771:5)
npm ERR!     at async Config.set (/usr/local/lib/node_modules/npm/lib/commands/config.js:176:5)
npm ERR!     at async Config.exec (/usr/local/lib/node_modules/npm/lib/commands/config.js:118:11)
npm ERR!     at async module.exports (/usr/local/lib/node_modules/npm/lib/cli-entry.js:61:5) {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/usr/local/etc'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in: /Users/oliclive-griffin/.npm/_logs/2025-02-26T00_53_59_710Z-debug-0.log
```

--- a/README.md
+++ b/README.md
@@ -17,11 +17,27 @@ Some of its key capabilities include:
 
 ## Get started
 
-1. Run the following command in your terminal: <br />
-  ```sh
-  npm install -g @anthropic-ai/claude-code
-  ```
-2. Navigate to your project directory and run <code>claude</code>
+1. Install Claude Code using one of these methods:
+   
+   **User-level installation** (recommended, no `sudo` required):
+   ```sh
+   mkdir -p ~/.local/share/npm-packages
+   npm config set prefix ~/.local/share/npm-packages
+   export PATH=~/.local/share/npm-packages/bin:$PATH
+   npm install -g @anthropic-ai/claude-code
+   ```
+   
+   *Add `export PATH=~/.local/share/npm-packages/bin:$PATH` to your `~/.bashrc` or `~/.zshrc` to make the change permanent.*
+   
+   **Installation inside of a virtual environment** (from `nvm`, etc...):
+   ```sh
+   npm install -g @anthropic-ai/claude-code
+   ```
+
+   *Do NOT install this package as an administrative user, such as `root`. This can compromise the security and stability of your system.*
+
+2. Navigate to your project directory and run <code>claude</code>.
+
 3. Complete the one-time OAuth process with your Anthropic Console account.
 
 ### Research Preview

--- a/README.md
+++ b/README.md
@@ -17,28 +17,21 @@ Some of its key capabilities include:
 
 ## Get started
 
-1. Install Claude Code using one of these methods:
+1. If you are new to Node.js and Node Package Manager (`npm`), then it is recommended that you configure an NPM prefix for your user.
+   Instructions on how to do this can be found [here](https://docs.anthropic.com/en/docs/agents-and-tools/claude-code/overview#recommended-create-a-new-user-writable-npm-prefix).
+
+   *Important* We recommend installing this package as a non-privileged user, not as an administrative user like `root`.
+   Installing as a non-privileged user helps maintain your system's security and stability.
+
+2. Install Claude Code:
    
-   **User-level installation** (recommended, no `sudo` required):
-   ```sh
-   mkdir -p ~/.local/share/npm-packages
-   npm config set prefix ~/.local/share/npm-packages
-   export PATH=~/.local/share/npm-packages/bin:$PATH
-   npm install -g @anthropic-ai/claude-code
-   ```
-   
-   *Add `export PATH=~/.local/share/npm-packages/bin:$PATH` to your `~/.bashrc` or `~/.zshrc` to make the change permanent.*
-   
-   **Installation inside of a virtual environment** (from `nvm`, etc...):
    ```sh
    npm install -g @anthropic-ai/claude-code
    ```
 
-   *Do NOT install this package as an administrative user, such as `root`. This can compromise the security and stability of your system.*
+3. Navigate to your project directory and run <code>claude</code>.
 
-2. Navigate to your project directory and run <code>claude</code>.
-
-3. Complete the one-time OAuth process with your Anthropic Console account.
+4. Complete the one-time OAuth process with your Anthropic Console account.
 
 ### Research Preview
 

